### PR TITLE
upgrade sbt to 1.2.6

### PR DESCRIPTION
**Pull request by Qi77Qi**
Maintenance work. None of the bullet points apply

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,9 +19,9 @@ before_install:
   - sh docker/run-opendj.sh start
   
 script:
-  - SBT_OPTS=-Xmx3g sbt clean coverage test -Denv.type=test -Dmysql.host=localhost -Dmysql.port=3307 -Ddirectory.url=ldap://localhost:3389 -Ddirectory.password=testtesttest it:compile
+  - SBT_OPTS=-J-Xmx3g sbt clean coverage test -Denv.type=test -Dmysql.host=localhost -Dmysql.port=3307 -Ddirectory.url=ldap://localhost:3389 -Ddirectory.password=testtesttest it:compile
 
 after_success:
-  - SBT_OPTS=-Xmx3g sbt coveralls
+  - SBT_OPTS=-J-Xmx3g sbt coveralls
```
Comment (helgridly) [inline .travis.yml:25]: does the `-J` change mean you have to change the docker build / run things too?

Comment (helgridly) [reply .travis.yml:25]: possibly also the docker-compose in fc-develop

Comment (Qi77Qi) [reply .travis.yml:25]: things are a bit weird here. I don't know the case for rawls yet. but in sam, I didn't find where the version for docker is defined, and it's still 0.13, which is why I can't add `-J` for docker file

File: project/plugins.sbt [hunk 2]
```diff
@@ -1,10 +1,10 @@
-addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.3")
+addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.7")
 
-addSbtPlugin("io.spray" % "sbt-revolver" % "0.8.0")
+addSbtPlugin("io.spray" % "sbt-revolver" % "0.9.0")
```
Comment (breilly2) [inline project/plugins.sbt]: Should be able to go to 0.9.1

File: project/Compiling.scala [hunk 3]
```diff
@@ -14,6 +14,6 @@ object Compiling {
   }
 
   val rawlsCompileSettings = List(
-    resourceGenerators in Compile <+= writeVersionConf
+    resourceGenerators in Compile += {writeVersionConf}
```
Comment (breilly2) [inline project/Compiling.scala]: When I tried this upgrade (and didn't really know what I was doing) I came up with:
```
Compile / resourceGenerators += writeVersionConf
```
Not even sure if that works or which is better. 🤷‍♂️

**Review by breilly2 — APPROVED** [review]
Just a couple of comments but generally 👍. Should probably bump up the sbt-revolver version, but you caught everything that I had when I experimented with this and then some.

One caveat: I don't really understand the travis changes, and I have the same question as Hussein about docker. We should probably understand this better.

File: .travis.yml (hunk 1 above)
Comment (helgridly) [reply .travis.yml:25]: in fc-develop [we pass in JAVA_OPTS](https://github.com/broadinstitute/firecloud-develop/blob/dev/run-context/live/configs/rawls/docker-compose.yaml.ctmpl#L31). And in the Dockerfile we just call `CMD java $JAVA_OPTS -jar rawls.jar`. So I think we're fine here.

**Review by helgridly — APPROVED** [review]

**Pull request merged**